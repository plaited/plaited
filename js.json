{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "plaited",
    "paths": {
      "plaited/jsx-runtime": [
        "./src/jsx/runtime.ts"
      ],
      "plaited/jsx-dev-runtime": [
        "./src/jsx/dev-runtime.ts"
      ]
    }
  }
}