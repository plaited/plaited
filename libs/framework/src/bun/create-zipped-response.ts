export const jsMimeTypes = ['text/javascript;charset=utf-8', 'application/json;charset=utf-8'] as const

export const isJsMimeType = (type: string): type is (typeof jsMimeTypes)[number] =>
  jsMimeTypes.includes(type as (typeof jsMimeTypes)[number])

export const createZippedResponse = (str: string, type: (typeof jsMimeTypes)[number]) => {
  const compressed = Bun.gzipSync(Buffer.from(str))
  return new Response(compressed, {
    headers: {
      'content-type': type,
      'content-encoding': 'gzip',
    },
  })
}
