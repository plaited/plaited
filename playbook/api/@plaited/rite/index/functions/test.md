**@plaited/rite** ( [Readme](../../README.md) \| API )

***

[Plaited Typedocs](../../../../modules.md) / [@plaited/rite](../../modules.md) / [index](../README.md) / test

# Function: test()

> **test**(`name`, `cb`): `void`

## Parameters

▪ **name**: `string`

▪ **cb**: `TestCallback`

## Returns

`void`

## Source

[libs/rite/src/test.ts:10](https://github.com/plaited/plaited/blob/0d4801d/libs/rite/src/test.ts#L10)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
