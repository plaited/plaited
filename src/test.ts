export * from './test/assert.types.js'
export * from './test/assert.constants.js'
export { assert } from './test/assert.js'
export { findByAttribute } from './test/find-by-attribute.js'
export { findByText } from './test/find-by-text.js'
export { fireEvent } from './test/fire-event.js'
export { match } from './test/match.js'
export { throws } from './test/throws.js'
export { wait } from './utils/wait.js'
export { AssertionError, MissingTestParamsError, TimeoutError } from './test/errors.js'
export { usePlay } from './test/use-play.js'
