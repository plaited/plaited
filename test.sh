#!/bin/bash

# Run yarn
yarn install --immutable

# Run yarn build
yarn build

# Run test
yarn test
