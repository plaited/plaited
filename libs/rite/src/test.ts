/// <reference lib="dom" />
/// <reference lib="dom.iterable" />

import { TestResultError, TestResult, BrowserSessionResult } from '@web/test-runner-core/browser/session.js'
import { assert, Assertion, AssertionError } from './assert.js'

type TestCallback = (t: Assertion) => Promise<void> | void

interface Test {
  (name: string, cb: TestCallback): void
  skip: (name: string, cb: TestCallback) => void
}

type TestCase = () => Promise<boolean> | boolean

class TestRunner {
  #completedTests = 0
  #failedTests = 0
  #errors: TestResultError[] = []
  #cases: TestCase[] = []
  #timeout = 5_000
  #tests: TestResult[] = []
  updateTimeout(time: number) {
    this.#timeout = time
  }
  get timeout() {
    return this.#timeout
  }
  addCase(test: TestCase) {
    this.#cases.push(test)
  }
  addResult(result: TestResult) {
    this.#tests.push(result)
  }
  addError(obj: TestResultError) {
    this.#errors.push(obj)
  }
  async run(name: string) {
    for (const testCase of this.#cases) {
      try {
        const passed = await testCase()
        this.#completedTests++
        if (!passed) {
          this.#failedTests++
        }
      } catch (error) {
        console.error(error)
      }
    }

    if (this.#completedTests === this.#cases.length) {
      const results: BrowserSessionResult = {
        passed: !this.#failedTests,
        testResults: {
          name,
          suites: [],
          tests: this.#tests,
        },
        errors: this.#errors,
      }
      return results
    } else {
      throw new AssertionError(`Not all test completed`)
    }
  }
}

declare global {
  interface Window {
    __rite_test_runner?: TestRunner
  }
}

window.__rite_test_runner = new TestRunner()

const throwTimeoutError = async () => {
  const timeout = window.__rite_test_runner.timeout
  await assert.wait(timeout)
  throw new Error(`test takes longer than ${timeout}ms to complete`)
}

export const test: Test = (name, callback) => {
  window.__rite_test_runner.addCase(async () => {
    const startTime = performance.now()
    const result: TestResult = {
      name,
      passed: true,
      skipped: false,
    }
    try {
      await Promise.race([callback(assert), throwTimeoutError()])
      const msg = `✓ ${name}`
      // eslint-disable-next-line no-console
      console.log(msg)
    } catch (error) {
      const msg = `✗ ${name}`
      // eslint-disable-next-line no-console
      console.log(msg)
      result.passed = false
      if (error instanceof AssertionError) {
        const obj: Pick<TestResultError, 'actual' | 'expected' | 'message'> = JSON.parse(error.message)
        result.error = { name, stack: `${error.stack}`, ...obj }
        window.__rite_test_runner.addError({ name, ...obj })
      } else {
        throw new Error(error)
      }
    }
    result.duration = performance.now() - startTime
    window.__rite_test_runner.addResult(result)
    return result.passed
  })
}
test.skip = (name: string, _: TestCallback) => {
  window.__rite_test_runner.addCase(() => {
    const result = {
      name,
      passed: true,
      skipped: true,
    }
    window.__rite_test_runner.addResult(result)
    return true
  })
}
