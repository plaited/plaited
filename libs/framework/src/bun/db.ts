import { Database } from "bun:sqlite";

export const db = new Database(":memory:");
db.query(`create table templates;`);