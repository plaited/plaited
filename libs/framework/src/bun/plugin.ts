import type {  BunPlugin } from "bun";
import { db } from "./db.js";

export const plugin: BunPlugin = {
  name: "Plaited loader",
  setup(build) {
    build.onResolve({ filter: /\/route.tsx?$/ }, async ({ path }) => {
      // mark external all non-template files
    })
    build.onLoad({ filter: /\/route.tsx?$/ }, async ({ path }) => {
      // modify the contents of the file to be an export object.
      return {
        contents,
        loader: "tsx",
      };
    });
  },
}