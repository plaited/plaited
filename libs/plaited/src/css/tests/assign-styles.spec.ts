import { test, expect } from 'bun:test'
import { assignStyles, createStyles, css } from '../index.js'

test('assignStyles', () => {
  const styles = createStyles({
    button: {
      fontFamily: 'Nunito Sans, Helvetica Neue, Helvetica, Arial, sans-serif',
      fontWeight: 700,
      border: 0,
      borderRadius: '3em',
      cursor: 'pointer',
      display: 'inline-block',
      lineHeight: 1,
    },
    primary: {
      color: 'white',
      backgroundColor: '#1ea7fd',
    },
    secondary: {
      color: '#333',
      backgroundColor: 'transparent',
      boxShadow: 'rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset',
    },
    small: {
      fontSize: '12px',
      padding: '10px 16px',
    },
    large: {
      fontSize: '16px',
      padding: '12px 24px',
    },
  })
  const host = css`
    :host {
      color: red;
    }
  `
  let primary = true
  expect(
    assignStyles(styles.button, styles['small'], primary ? styles.primary : styles.secondary, host),
  ).toMatchSnapshot()
  primary = false
  expect(
    assignStyles(styles.button, styles['large'], primary ? styles.primary : styles.secondary, host),
  ).toMatchSnapshot()
})
