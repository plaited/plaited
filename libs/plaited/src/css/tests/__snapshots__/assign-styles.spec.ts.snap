// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`assignStyles 1`] = `
{
  "className": "button_kvg5vg p49uqw9n p1ri95dn p-c0syh9 p-2ezhu34 p-2qvsfi4 p2plywwi p10xl1ef small_163g06v pntfuxu p12x8yyc primary_-156y5z6 pyird39 p-2w4nzkx",
  "stylesheet": [
    ".p49uqw9n{font-family:Nunito Sans, Helvetica Neue, Helvetica, Arial, sans-serif}",
    ".p1ri95dn{font-weight:700}",
    ".p-c0syh9{border:0}",
    ".p-2ezhu34{border-radius:3em}",
    ".p-2qvsfi4{cursor:pointer}",
    ".p2plywwi{display:inline-block}",
    ".p10xl1ef{line-height:1}",
    ".pntfuxu{font-size:12px}",
    ".p12x8yyc{padding:10px 16px}",
    ".pyird39{color:white}",
    ".p-2w4nzkx{background-color:#1ea7fd}",
  ],
}
`;

exports[`assignStyles 2`] = `
{
  "className": "button_kvg5vg p49uqw9n p1ri95dn p-c0syh9 p-2ezhu34 p-2qvsfi4 p2plywwi p10xl1ef large_10r87xp p14vlh7q p188r5x1 secondary_49l08bw p-f5f5uo p1ti1rrq pqxf18d",
  "stylesheet": [
    ".p49uqw9n{font-family:Nunito Sans, Helvetica Neue, Helvetica, Arial, sans-serif}",
    ".p1ri95dn{font-weight:700}",
    ".p-c0syh9{border:0}",
    ".p-2ezhu34{border-radius:3em}",
    ".p-2qvsfi4{cursor:pointer}",
    ".p2plywwi{display:inline-block}",
    ".p10xl1ef{line-height:1}",
    ".p14vlh7q{font-size:16px}",
    ".p188r5x1{padding:12px 24px}",
    ".p-f5f5uo{color:#333}",
    ".p1ti1rrq{background-color:transparent}",
    ".pqxf18d{box-shadow:rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset}",
  ],
}
`;
